### PR TITLE
Doctors views

### PR DESCRIPTION
**Pull request by ileanahi**
Created Doctors pages
- Doctor's profile
- List of Doctors

### CODE REVIEW

File: src/main/java/com/capstone/doctorsOffice/controllers/DoctorViewController.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    @Autowired
+    private PatientRepository patientRepository;
+
+    @Autowired
+    private AppointmentRepository appointmentRepository;
+
+    @GetMapping("/{id}")
+    public String getDoctorById(@PathVariable Long id, Model model) throws ChangeSetPersister.NotFoundException {
+        System.out.println("**************** doctor view!!! ** -------------------");
```
Comment (williamsc51) [inline src/main/java/com/capstone/doctorsOffice/controllers/DoctorViewController.java]: This is not a big deal and you can choose to leave them if you like. But all the `System.out.println` I'm assuming is for debugging purpose. We wouldn't want to commit debug code into the repository.
Ideally we would remove all the `System.out.println`

File: src/main/java/com/capstone/doctorsOffice/controllers/DoctorViewController.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+        model.addAttribute("doctor", doctor);
+        model.addAttribute("patients", patients);
+        model.addAttribute("appointments", appointments);
+
+        return "doctor-profile";
+    }
+
+    @GetMapping("/")
```
Comment (williamsc51) [inline src/main/java/com/capstone/doctorsOffice/controllers/DoctorViewController.java]: remove `"/"` let it just be an empty string `""`
With the current settings the url only works with the `/` at the end `http://localhost:8080/doctorView/`
The more natural url to list a collection is `http://localhost:8080/doctorView`

File: src/main/resources/templates/doctors.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+<div class="title-header">
+    <p>Doctors</p>
+</div>
+
+<div class="content">
+
+    <div class="patient-info">
+        <p><b>Doctors:</b></p>
+        <tr th:if="${doctors.isEmpty()}">
```
Comment (williamsc51) [inline src/main/resources/templates/doctors.html]: missing `table` tag. `tr` and `td` must be inside of a table tag.

The list is not rending correctly when displaying more that one doctor.
Screen Shot 2023-02-12 at 6 54 55 PM

File: src/main/resources/templates/doctor-profile.html [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+<div class="content">
+
+    <div class="patient-info">
+        <p><b>Profession:</b></p>
+        <p><span th:if="${doctor.isPresent()}" th:text="${doctor.get().profession}"></span></p>
+    </div>
+
+    <div class="patient-info">
+        <p><b>Next Appointment:</b></p>
```
Comment (williamsc51) [inline src/main/resources/templates/doctor-profile.html]: Opinion:
If this is from the patient perspective, we do not want to display the doctors appointments and what patient the appointment is with to all patients.

We also should not display the doctor's patient list.
We should keep this page simple only displaying the doctor name and profession.

**Review by williamsc51 — CHANGES_REQUESTED** [review]

File: src/main/resources/templates/index.html [hunk 5]
```diff
@@ -16,7 +16,7 @@
         <ul>
             <li><a href="../">Home</a></li>
             <li><a href="../">Patient Portal</a></li>
-            <li><a href="../">Doctor Portal</a></li>
+            <li><a th:href="@{'/doctorView/'}">Doctor Portal</a></li>
             <li><a href="../">Pharmacy</a></li>
```
Comment (williamsc51) [inline src/main/resources/templates/index.html:20]: remove trailing `/`
```
@{'/doctorView'}
```

File: src/main/resources/templates/patient-profile.html [hunk 6]
```diff
@@ -16,7 +16,7 @@
             <ul>
                 <li><a href="../">Home</a></li>
                 <li><a href="../">Patient Portal</a></li>
-                <li><a href="../">Doctor Portal</a></li>
+                <li><a th:href="@{'/doctorView/'}">Doctor Portal</a></li>
                 <li><a href="../">Pharmacy</a></li>
```
Comment (williamsc51) [inline src/main/resources/templates/patient-profile.html:20]: remove trailing `/`
```
@{'/doctorView'}
```

File: src/main/java/com/capstone/doctorsOffice/controllers/DoctorViewController.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+    @GetMapping("/{id}")
+    public String getDoctorById(@PathVariable Long id, Model model) throws ChangeSetPersister.NotFoundException {
+        Optional<DoctorDto> doctor = doctorService.getDoctorById(id);
+        List<Patient> patients = patientRepository.findByDoctorId(id);
+        List<Appointment> appointments = appointmentRepository.findByDoctorId(id);
+
+        model.addAttribute("doctor", doctor);
+        model.addAttribute("patients", patients);
+        model.addAttribute("appointments", appointments);
```
Comment (williamsc51) [inline src/main/java/com/capstone/doctorsOffice/controllers/DoctorViewController.java]: remove lines 42 to 46.
We are no longer using those in the views. 
Remove lines 34 and 37 as well.

File: src/main/resources/templates/doctors.html [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    <div class="patient-info">
+        <p><b>Doctors:</b></p>
+        <p th:if="${doctors.isEmpty()}">
+            <td>No doctors.</td>
+        </p>
+        <table>
+            <tr th:unless="${doctors.isEmpty()}"  th:each="doctor : ${doctors}">
+                <a th:href="@{'/doctorView/' + ${doctor.id}}">
+                    <td th:text="${doctor?.name} + ' - ' + ${doctor?.profession}"></td>
+                </a>
```
Comment (williamsc51) [inline src/main/resources/templates/doctors.html]: The a tag have to be inside the td tag for it to work.
should be:
```
                    <td>
                        <a th:href="@{'/doctorView/' + ${doctor.id}}">
                            <span th:text="${doctor?.name} + ' - ' + ${doctor?.profession}"></span>
                        </a>
                    </td>
```

**Review by williamsc51 — CHANGES_REQUESTED** [review]

**Review by asya-code — APPROVED** [review]

**Review by GRISONRF — APPROVED** [review]

**Review by mlandes-hackbright — APPROVED** [review]

**Pull request merged**